compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/AbstractTypeSymbol.java

```diff
@@ -102,6 +102,11 @@ public boolean assignableTo(TypeSymbol targetType) {
         return types.isAssignable(this.bType, getTargetBType(targetType));
     }
 
+    @Override
+    public boolean nameEquals(String name) {
+        return false;
```

Comment:
This will have to be overridden in some type symbols. e.g., type ref, builtin subtypes

Let's maybe go with this for now as the initial implementation and address it in a separate PR.

Reply:
I kept this like this because I didn't see any requirement for alternative implementation in built-in subtypes. Do we have any?

Reply:
I mean the subtypes of `AbstractTypeSymbol`

Reply:
Consider a type ref type symbol. Those are named. But this will always return `false` for type ref type symbols too. Same goes for other type symbols in which the `getName()` method implementation of `AbstractTypeSymbol` is overridden.

Reply:
Ah got the idea :+1:

Reply:
Let's add this change. I'll update the PR